### PR TITLE
Make Kubernetes control plane and etcd scheduling configurable

### PR DESCRIPTION
**Pull request by joshwget**
cc @sangeethah @aemneina @deniseschannon @alena1108

### CODE REVIEW

File: infra-templates/k8s/20/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,230 @@
+kubelet:
+    labels:
+        io.rancher.container.dns: "true"
+        io.rancher.container.create_agent: "true"
+        io.rancher.container.agent.role: environmentAdmin
+        io.rancher.scheduler.global: "true"
+        io.rancher.scheduler.affinity:host_label_ne: nopods=true
```
Comment (joshwget) [inline infra-templates/k8s/20/docker-compose.yml]: The label for `proxy` and `kubelet` were previously a hard affinity. Should these also be configurable or remain hard always?

File: infra-templates/k8s/20/rancher-compose.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    type: int
+  - variable: ETCD_ELECTION_TIMEOUT
+    label: Etcd Election Timeout
+    description: Time (in milliseconds) for an election to timeout.
+    required: true
+    default: 5000
+    type: int
+  - variable: DEDICATED_HOSTS
+    label: Dedicated hosts
+    description:  Run the Kubernetes control plane and etcd on decicated hosts
```
Comment (joshwget) [inline infra-templates/k8s/20/rancher-compose.yml]: Is this a clear description?

**Comment by joshwget** [comment]
It looks like we validate `docker-compose.yml` in CI. This doesn't mix well with templating.

File: infra-templates/k8s/20/rancher-compose.yml (hunk 2 above)
Comment (deniseschannon) [reply infra-templates/k8s/20/rancher-compose.yml]: There's a typo, but I think that's fine. I'm not sure if we need to give more details on the labels of the hosts or maybe a link to the docs (when I publish them)?

Comment (joshwget) [reply infra-templates/k8s/20/rancher-compose.yml]: Linking to the docs is a good idea. Maybe we could add that in the README for the template.

**Comment by joshwget** [comment]
Fixed typo, moved back to version 19.

**Comment by LLParse** [comment]
This is 95% correct, I just made a couple tweaks and got rid of the nopods concept which was erroneous logic https://github.com/rancher/rancher-catalog/pull/487

Also managed to avoid the simple CI validation in this case, will need to brainstorm with @cloudnautique on how we can adjust that validation to handle the templating.

**Comment by joshwget** [comment]
Closing in favor of #487. Thanks @LLParse!

**Comment by LLParse** [comment]
Haha, I posted wrong PR, for linking purposes this is it https://github.com/rancher/rancher-catalog/pull/491

**Pull request closed without merging**